addons/mail/models/discuss/ir_binary.py

```diff
@@ -0,0 +1,12 @@
+from odoo import models
+
+
+class IrBinary(models.AbstractModel):
+    _inherit = 'ir.binary'
+
+    def _find_record_check_access(self, record, access_token, field):
+        if record._name in ["res.partner", "mail.guest"] and field == "avatar_128":
+            guest = self.env["mail.guest"]._get_guest_from_context()
+            if self.env.user._is_public() and guest and guest.sudo().channel_ids & record.sudo().channel_ids:
```

Comment:
Also I don't like we use `record.channel_ids` in a generic way, relying on the fact that the same of the field is the same on the partner and guest model.